File: src/algorithms/is-palindrome.js
/**
 * Returns true if s is a palindrome, or false otherwise.
 * 
 * A phrase is a palindrome if, after converting all uppercase letters into 
 * lowercase letters and removing all non-alphanumeric characters, it reads the 
 * same forward and backward. Alphanumeric characters include letters and 
 * numbers.
 * 
 * @param {string} s - A string of printable ASCII characters
 * @returns {boolean} - Returns true if s is a palindrome
 */
function isPalindrome (s) {
  let res = '';
  s = s.toLowerCase();
  for (let i = 0; i < s.length; i++) {
    let curr = s.charCodeAt(i);
    if ((curr >= 97 && curr <= 122) || (curr >= 48 && curr <= 57)) {
      res += s[i];
    }
  }
  let reverse = res.split('').reverse().join('');
  return res === reverse;
}

File: src/algorithms/is-palindrome.py
def is_palindrome(x):
    """Given an integer x, returns true if x is a palindrome.

    Args:
        x (int): an integer

    Returns:
        Boolean: returns true if x is a palindrome
    """
    if x < 0 or (x % 10 == 0 and x != 0):
        return False
    reverse = 0
    while x > reverse:
        digit = x % 10
        reverse = reverse * 10 + digit
        x //= 10
    return x == reverse or x == reverse // 10
